esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -300,25 +300,29 @@ bool CWsESDLConfigEx::existsESDLMethodDef(const char * esdlDefinitionName, unsig
         IPropertyTree * esxdl = globalLock->queryRoot();
         if (esxdl)
         {
-            VStringBuffer mxpath("EsdlService[@name='%s']", esdlServiceName);
-
             Owned<IPropertyTreeIterator> it = esxdl->getElements("EsdlService");
+            int servicesCount = esxdl->getCount("EsdlService");
+
             ForEach(*it)
             {
-                IPropertyTree* pChildNode = &it->query();
-                if (stricmp(pChildNode->queryProp("@name"), esdlServiceName)==0)
+                IPropertyTree* pCurrService = &it->query();
+                if ((servicesCount == 1 && !esdlServiceName.length()) || stricmp(pCurrService->queryProp("@name"), esdlServiceName.str())==0)
                 {
                     Owned<IPropertyTreeIterator> it2 = esxdl->getElements("EsdlMethod");
                     ForEach(*it2)
                     {
                         IPropertyTree* pChildNode = &it2->query();
                         if (stricmp(pChildNode->queryProp("@name"), methodName)==0)
                         {
+                            if (!esdlServiceName.length())
+                                esdlServiceName.set(pCurrService->queryProp("@name"));
```

Comment:
I think the conditional at 309 prevents that scenario; we only enter this block if either esdlServiceName == @name, or esdlService is empty and there's only one service...